odoo/modules/registry.py

```diff
@@ -467,30 +467,28 @@ def check_indexes(self, cr, model_names):
         if not expected:
             return
 
-        trgm = sql.has_pg_trgm(cr)
         cr.execute("SELECT indexname FROM pg_indexes WHERE indexname IN %s",
                    [tuple(row[0] for row in expected)])
         existing = {row[0] for row in cr.fetchall()}
 
-        for indexname, model, columnname, field in expected:
-            tablename = model._table
-            index = field.index
-            assert index in ('btree', 'gin', 'not null', True, False)
-            if field.index and indexname not in existing:
-                where = ''
+        if any(row[3] == 'trigram' for row in expected) and not self.has_trigram:
+            self.has_trigram = sql.install_pg_trgm(cr)
```

Comment:
`self.has_trigram` is already bind in `init` of the registry (coming from https://github.com/odoo/odoo/pull/65871) (it seems a lit bit different). Should we keep this one instead of check at every module installation ?